include/pybind11/pytypes.h

```diff
@@ -2270,16 +2267,9 @@ iterator object_api<D>::end() const {
     return iterator::sentinel();
 }
 template <typename D>
-item_accessor object_api<D>::operator[](handle key) const {
-    return {derived(), reinterpret_borrow<object>(key)};
-}
-template <typename D>
-item_accessor object_api<D>::operator[](object &&key) const {
-    return {derived(), std::move(key)};
-}
-template <typename D>
-item_accessor object_api<D>::operator[](const char *key) const {
-    return {derived(), pybind11::str(key)};
+template <typename T>
+item_accessor object_api<D>::operator[](T &&key) const {
+    return {derived(), detail::object_or_cast(std::forward<T>(key)).template cast<object>()};
```

Comment:
I dug around and found it that going from handle to object this way automatically does reinterpret_borrow so it's just some syntactic sugar in that regard.

Reply:
@Skylion007 sorry to bother you but is this "syntactic sugar" of @woodychow 's original code or does your code have improvement?